tiled/client/node.py

```diff
@@ -543,6 +549,87 @@ def _ipython_key_completions_(self):
             # Do not print messy traceback from thread. Just fail silently.
             return []
 
+    def create_array(self, array, metadata=None, specs=None):
+        """
+        EXPERIMENTAL: write an array
+
+        This is subject to change or removal without notice
+
+        """
+
+        from ..structures.array import ArrayMacroStructure, ArrayStructure, BuiltinDtype
+
+        metadata = metadata or {}
+        specs = specs or []
+        mimetype = "application/octet-stream"
+
+        structure = ArrayStructure(
+            macro=ArrayMacroStructure(
+                shape=array.shape,
+                # just one chunk for now...
+                chunks=tuple((size,) for size in array.shape),
+            ),
+            micro=BuiltinDtype.from_numpy_dtype(array.dtype),
+        )
+        data = {
+            "metadata": metadata,
+            "structure": asdict(structure),
+            "structure_family": StructureFamily.array,
+            "specs": specs,
+            "mimetype": mimetype,
+        }
+        document = self.context.post_json("/node/metadata/", data)
+        uid = document["uid"]
+        self.context.put_content(f"/array/full/{uid}", content=array.tobytes())
+
+    def create_dataframe(self, dataframe, metadata=None, specs=None):
+        """
+        EXPERIMENTAL: write a dataframe
+
+        This is subject to change or removal without notice
+
+
+        """
+        from dask.dataframe.utils import make_meta
+
+        from ..structures.dataframe import (
+            DataFrameMacroStructure,
+            DataFrameMicroStructure,
+            DataFrameStructure,
+        )
+
+        metadata = metadata or {}
+        specs = specs or []
+        mimetype = APACHE_ARROW_FILE_MIME_TYPE
+
+        structure = DataFrameStructure(
+            micro=DataFrameMicroStructure(meta=make_meta(dataframe), divisions=[]),
+            macro=DataFrameMacroStructure(
+                npartitions=1, columns=list(dataframe.columns)
+            ),
+        )
+
+        data = {
+            "metadata": metadata,
+            "structure": asdict(structure),
+            "structure_family": StructureFamily.dataframe,
+            "specs": specs,
+            "mimetype": mimetype,
+        }
+
+        data["structure"]["micro"]["meta"] = base64.b64encode(
+            bytes(serialize_arrow(data["structure"]["micro"]["meta"], {}))
+        ).decode()
+
+        document = self.context.post_json("/node/metadata/", data)
+        uid = document["uid"]
+
+        write_buffer = BytesIO()
+        dataframe.to_csv(write_buffer)
```

Comment:
Does this need to be Arrow, not CSV? The `serialize_arrow` function would be useful here.